src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+#if DNX451
+using System.Runtime.Remoting.Messaging;
+using System.Runtime.Remoting;
+#elif DNXCORE50
+using System.Threading;
+#endif
+
+namespace Microsoft.AspNet.Http
```

Comment:
The internal namespace is for framework infrastructure that's probably not used by the general public.
Compare to DefaultHttpContext which Hosting also uses: https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http/DefaultHttpContext.cs#L15